src/navigation.ts

```diff
@@ -432,6 +440,34 @@ export class Navigation {
     }
   }
 
+  /**
+   * Handle the widget or dropdown div losing focus (via focusout).
+   *
+   * Because we skip the widget/dropdown div cases in `handleBlurWorkspace` we need
+   * to catch them here.
+   *
+   * @param workspace The workspace.
+   * @param relatedTarget The related target (newly focused element if any).
+   */
+  handleFocusOutWidgetDropdownDiv(
+    workspace: Blockly.WorkspaceSvg,
+    relatedTarget: EventTarget | null,
+  ) {
+    if (relatedTarget === null) {
+      // Workaround:
+      // Skip document.body/null case until these blur bugs are fixed to avoid
+      // flipping to passive focus as the user moves their mouse over the
+      // dropdown/widget at the cost of clicks on body showing the wrong focus
+      // style.
+      // https://github.com/google/blockly-samples/issues/2498
+      // https://github.com/google/blockly/issues/8819
+      return;
```

Comment:
Update now Blockly is upgraded fixing one of the bugs.
We could also just drop this but it's very weird to experience in the demo with the colour picker.

```suggestion
      // Workaround:
      // Skip document.body/null case until this blur bugs is fixed to avoid
      // flipping to passive focus as the user moves their mouse over the
      // colour picker.
      // https://github.com/google/blockly-samples/issues/2498
      return;
```